src/pp_solvers/solver_mixin.py

```diff
@@ -84,7 +84,7 @@ def scale_energy_transform(J, row_groups: list[int], model: pp.PorePyModel):
     if len(subdomains) == 0:
         # No subdomains, hence no scaling.
         return Q
-    Q[row_groups] = sps.diags(vols, format="csr")
+    Q.set_diagonal(groups=row_groups, values=vols, additive=False)
```

Comment:
The original line demanded 300 Gigs of memory for the 3D benchmark, case 4. Now it does not cost anything.

Reply:
Oh, that must be sps.diags vs sps.dia_matrix. It is not the first time I have encountered that. Thanks.